lib/models/addon.js

```diff
@@ -878,9 +937,22 @@ var Addon = CoreObject.extend({
     @return {Tree} Processed javascript file tree
   */
   processedAddonJsFiles: function(tree) {
-    return this.preprocessJs(this.addonJsFiles(tree), '/', this.name, {
-      registry: this.registry
-    });
+    if (this._fileSystemInfo().hasJSFiles) {
+      var processedJsFiles = this.preprocessJs(this.addonJsFiles(tree), '/', this.name, {
+        registry: this.registry
+      });
+
+      if (!registryHasPreprocessor(this.registry, 'js')) {
+        this._warn('Addon files were detected in `' + this._treePathFor('addon') + '`, but no JavaScript ' +
+                   'preprocessors were found for `' + this.name + '`. Please make sure to add a preprocessor ' +
+                   '(most likely `ember-cli-babel`) to in `dependencies` (NOT `devDependencies`) in ' +
+                   '`' + this.name + '`\'s `package.json`.');
```

Comment:
as it may be possible to have more then 1 addon by this name, should we also print its path on disk?